backend/internal/branding/resolve/service.go

```diff
@@ -0,0 +1,119 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package brandingresolve
+
+import (
+	"strings"
+
+	"github.com/asgardeo/thunder/internal/application"
+	"github.com/asgardeo/thunder/internal/branding/common"
+	brandingmgt "github.com/asgardeo/thunder/internal/branding/mgt"
+	"github.com/asgardeo/thunder/internal/system/error/serviceerror"
+	"github.com/asgardeo/thunder/internal/system/log"
+)
+
+const (
+	serviceLogger  = "BrandingResolveService"
+	resolveTypeAPP = "APP"
+)
+
+// BrandingResolveServiceInterface defines the interface for the branding resolve service.
+type BrandingResolveServiceInterface interface {
+	ResolveBranding(resolveType, id string) (*common.BrandingResponse, *serviceerror.ServiceError)
+}
+
+// brandingResolveService is the default implementation of the BrandingResolveServiceInterface.
+type brandingResolveService struct {
+	brandingMgtService brandingmgt.BrandingMgtServiceInterface
+	applicationService application.ApplicationServiceInterface
+	logger             *log.Logger
+}
+
+// newBrandingResolveService creates a new instance of BrandingResolveService with injected dependencies.
+func newBrandingResolveService(
+	brandingMgtService brandingmgt.BrandingMgtServiceInterface,
+	applicationService application.ApplicationServiceInterface,
+) BrandingResolveServiceInterface {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, serviceLogger))
+	return &brandingResolveService{
+		brandingMgtService: brandingMgtService,
+		applicationService: applicationService,
+		logger:             logger,
+	}
+}
+
+// ResolveBranding resolves a branding configuration by type and ID.
+// TODO: Add support for OU type and fallback logic.
+func (brs *brandingResolveService) ResolveBranding(
+	resolveType, id string,
+) (*common.BrandingResponse, *serviceerror.ServiceError) {
+	resolveTypeUpper := strings.ToUpper(resolveType)
+
+	if resolveTypeUpper == "" {
+		return nil, &common.ErrorInvalidResolveType
+	}
+
+	if id == "" {
+		return nil, &common.ErrorMissingResolveID
+	}
+
+	// Currently only APP type is supported
+	if resolveTypeUpper != resolveTypeAPP {
+		return nil, &common.ErrorUnsupportedResolveType
+	}
+
+	// Get the application by ID
+	if brs.applicationService == nil {
+		brs.logger.Error("Application service is not available")
+		return nil, &serviceerror.InternalServerError
+	}
+
+	app, svcErr := brs.applicationService.GetApplication(id)
+	if svcErr != nil {
+		// Convert application service errors to branding resolve errors
+		if svcErr.Code == "APP-1001" || svcErr.Code == "APP-1004" {
+			return nil, &common.ErrorApplicationNotFound
+		}
+		return nil, svcErr
+	}
+
+	// Check if the application has a branding ID
+	if app.BrandingID == "" {
+		return nil, &common.ErrorApplicationHasNoBranding
+	}
+
+	// Get the branding configuration
+	brandingConfig, svcErr := brs.brandingMgtService.GetBranding(app.BrandingID)
+	if svcErr != nil {
+		return nil, svcErr
```

Comment:
IMO we should catch and handle branding not found error as a server error here. 
For the management service, it is a valid scenario to return branding not found. But when an application has a non existing branding id associated, it's a cleanup issue and resolver should treat it as a server error.
WDYT?

If we're changing it to be a server error, make sure to log it as well